callouts/java/service-callout/src/main/java/service/ServiceCallout.java

```diff
@@ -0,0 +1,283 @@
+
+import java.io.IOException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static utils.SslUtils.createSslContext;
+import static utils.SslUtils.readFileToBytes;
+
+/**
+ * ServiceCallout is an abstract class representing a service callout server that can handle various stages of HTTP request/response processing.
+ */
+public abstract class ServiceCallout {
+    private static final Logger logger = Logger.getLogger(ServiceCallout.class.getName());
+
+    private Server server;
+
+    public ServiceCallout() {
+        this(null, null, null, null, null, null, null, null, null, null, null, null, null);
+    }
+
+    public ServiceCallout(
+            String ip,
+            Integer port,
+            Integer insecurePort,
+            String healthCheckIp,
+            Integer healthCheckPort,
+            String healthCheckPath,
+            Boolean serperateHealthCheck,
+            byte[] cert,
+            String certPath,
+            byte[] certKey,
+            String certKeyPath,
+            Integer serverThreadCount,
+            Boolean enableInsecurePort) {
+        if (ip != null) {
+            this.ip = ip;
+        }
+        if (port != null) {
+            this.port = port;
+        }
+        if (insecurePort != null) {
+            this.insecurePort = insecurePort;
+        }
+        if (healthCheckIp != null) {
+            this.healthCheckIp = healthCheckIp;
+        }
+        if (healthCheckPort != null) {
+            this.healthCheckPort = healthCheckPort;
+        }
+        if (healthCheckPath != null) {
+            this.healthCheckPath = healthCheckPath;
+        }
+        if (serperateHealthCheck != null) {
+            this.serperateHealthCheck = serperateHealthCheck;
+        }
+        if (certPath != null) {
+            this.certPath = certPath;
+        }
+
+        if (cert != null) {
+            this.cert = cert;
+        } else {
+            this.cert = readFileToBytes(this.certPath);
+        }
+
+        if (certKeyPath != null) {
+            this.certKeyPath = certKeyPath;
+        }
+
+        if (certKey != null) {
+            this.certKey = certKey;
+        } else {
+            this.certKey = readFileToBytes(this.certKeyPath);
+        }
+
+        if (serverThreadCount != null) {
+            this.serverThreadCount = serverThreadCount;
+        }
+        if (enableInsecurePort != null) {
+            this.enableInsecurePort = enableInsecurePort;
+        }
+    }
+
+    private String ip = "0.0.0.0";
+    private int port = 8443;
+    private int insecurePort = 8443;
+    private String healthCheckIp = "0.0.0.0";
+    private int healthCheckPort = 8000;
+    private String healthCheckPath = "/";
+    private boolean serperateHealthCheck = false;
+    private byte[] cert = null;
+    private String certPath = "certs/server.crt";
+    private byte[] certKey = null;
+    private String certKeyPath = "certs/pkcs8_key.pem";
```

Comment:
Do you know where this key came from? If not we should remove it and for now I think we can copy the python certs.

Reply:
We should probably combine them, but I am unsure the best way to do this right now. We may need to modify the docker configurations. Or create them on the fly.

Reply:
Done